Onelogin OIDC (#1392)
spectrocloud/librarium: content/docs/08-user-management/04-saml-sso/05-palette-sso-with-onelogin.md

<br />
  
  ![Add Security Policy](/oidc-onelogin-images/group-sec-policy.png)

<br />

15. For this example we will automate the mapping of a user to a role and group, by creating a Mapping. Go to **Users** --> **Mappings** and select **New Mapping**. We'll give the mapping a name and set it to map every member of the *Administrators* group to automatically get the *Admin* role and set the **MemberOf** value to *Administrators*. The latter is essential so the response from OneLogin contains the group name, which we will match with a Team in Palette. Without explicitly setting the *MemberOf* value, Palette will not get the group name and won't be able to set the correct RBAC settings for your user. 

<br />

  ![Add Mapping](/oidc-onelogin-images/mapping-role-group.png)

<br />

16. Go to **Users** --> **Users** and select your user. Select the **Authentication** tab and select the group and security policy you just created. Click Save. 

<br />

  ![Add User to Group](/oidc-onelogin-images/user-auth-group.png)

<br />

## Enable OIDC in Palette 

> 17. You're done configuring OneLogin, now you can navigate back to Palette and fill in the **Client ID**, **Client Secret** and **Issuer URL** values. Also add the **groups** scope to the **Scopes** field. Without this Palette won't receive the group name. With everything filled in, you can now click **Enable**. 

Reviewer: ```suggestion
17. Navigate back to Palette and fill in the **Client ID**, **Client Secret**, and **Issuer URL** values. Add the **groups** scope to the **Scopes** field. Without the proper scopes, Palette won't receive the group name. Click **Enable** to continue. 
```

<br />

  ![Full OIDC Config](/oidc-onelogin-images/oidc-full-palette.png)

<br />

## Creating Teams in Palette

> 18. In order for Palette to map the groups configured in OneLogin, you need to create a Team in Palette. This Team name needs to match with the value that is returned for the groups configured in OneLogin. In this example we'll be creating an **Administrators** team. This also matches with the Administrators value that is assigned to each member of the Administrators group. In Palette, go to **Tenant Settings** --> **Users & Teams** --> **Teams** and select **Create team**. You only need to fill in the Team name and don't have to manually add members. This will happen automatically, because the users will belong to a certain group and that group is mapped to a team with certain permissions in Palette. After filling in the Team name, you can click *confirm*. 

Reviewer: ```suggestion
18. In order for Palette to map the groups configured in OneLogin, you need to create a Team in Palette. This Team name needs to match the value that is returned for the groups configured in OneLogin. In this example, you will create an **Administrators** team. This also matches with the Administrators value that is assigned to each member of the Administrators group in OneLogin. In Palette, navigate to left **Main Menu**, select **Tenant Settings** --> **Users & Teams** --> **Teams**, and select **Create team**. You only need to fill in the Team name. Don't manually add members, as OneLogin will manage the users. In OneLogin, users who are part of the mapped group, administrators, will be automatically added to the team. Click **Confirm** to continue. 
```
Language cleanup.

<br />

  ![Create New Team](/oidc-onelogin-images/new-team.png)

<br />

> 19. Next you need to give the members of this team a set of permissions. In this example we will give all members *Tenant Admin* permissions, since this is the Administrators team. Of course, you can define these permissions fully to your liking. In order to set the correct permissions, you can select the team you just created (**Tenant Settings** --> **Users & Teams** --> **Teams** --> **Your team**), then select **Tenant Roles** --> **Add Tenant Role** and select the **Tenant Admin** box. This will auto-select all other boxes. 

Reviewer: ```suggestion
19. Next, you need to give the members of this team a set of permissions. Assign all members *Tenant Admin* permissions because this is the Administrators team. You can define permissions fully to your liking. To set the correct permissions, navigate to left **Main Menu**, select **Tenant Settings** --> **Users & Teams** --> **Teams**. Select the team you created. Click on **Tenant Roles** --> **Add Tenant Role** and select the **Tenant Admin** box. This will auto-select all other permission boxes.  Click **Confirm**.
```

<br />

  ![Add Tenant Role](/oidc-onelogin-images/tenant-admin-role.png)

<br />

Click **Confirm** and you should have a configuration similar to this. 

<br />

  ![Permissions](/oidc-onelogin-images/roles-full.png)

<br />

> 

Reviewer: ```suggestion
You have now configured Palette to use OneLogin as a third-party IDP. Use the above steps to create additional groups in OneLogin and Palette. As you start deploying clusters, keep in mind that if you use Palette's Extended Kubernetes distribution, you can configure the cluster to inherit OIDC configurations from the tenant. Refer to the [Palette eXtended Kubernetes](/integrations/kubernetes#configureoidcidentityprovider) reference page to learn more about out-of-the-box support for OIDC.
```

I think we should have some sort of closing statement. Feel free to change or use the above suggestion.

# Validate

> 1. The SSO configuration is now complete and it's time to test all of our efforts. You can logout out of Palette and you'll be redirected to a new login screen. 

Reviewer: ```suggestion
Use the following steps to validate the configuration.

<br /

1. Log out of Palette. You will be redirected to a new login screen.
```
Changed to more formal language. The validation steps should not include the intro, so I placed that above.

<br />

  ![Logout](/oidc-onelogin-images/logout.png)

<br />

> 2. The login screen now defaults to a **Sign In** button. If you are already signed in into OneLogin, once you click on **Sign In** you will be automatically signed in and should have the correct permissions set. If you are unable to sign in, you can always use a username and password. 

Reviewer: ```suggestion
2. The login screen now defaults to a **Sign In** button. Click on **Sign In**. You will be required to authenticate with OneLogin. If you are already authenticated with OneLogin you will be automatically signed in to Palette with the proper permissions inherited from the respective Palette team you are a member of. If you cannot sign in, you can use the username and password method. 
```
Minor clarification and language touches.

<br />

  ![SSO Login Screen](/oidc-onelogin-images/login-screen.png)

<br />

> 3. You can check which teams you've been mapped to, by going to **Tenant Settings** --> **Users & Teams** --> **Users** and you'll directly see which teams you belong to.  

Reviewer: ```suggestion
3. You can check what teams you are mapped to by navigating to the left **Main Menu** and selecting **Tenant Settings**. Select **Users & Teams** --> **Users**. Find your user name to review the teams you are a member of. 
```

<br />

>   ![Check Team Member](/oidc-onelogin-images/team-member.png)

Reviewer: This screenshot is easy to miss. Let's include the entire screen but blur out other names.  That way, the reader can see where in the Palette UI to see this.

<br />

> # Troubleshooting

Reviewer: We don't have Troubleshooting in our style guide. You can add this as a paragraph or use an InfoBox.

> With the OpenID Connect Inspector, you can send requests to OneLogin and check what is sent in the payload. This way you can make sure that you are using the right claims and scopes. When using this, you will need to add the Inspector callback URL to your applications Redirect URIs list. 
> [OpenID Connect Inspector Tutorial](https://youtu.be/do0agd71hE8)
> [OpenID Connect Inspector](https://developers.onelogin.com/openid-connect/inspector)

Reviewer: ```suggestion
If you encounter issues while setting up OIDC, you can use OneLogin's [OpenID Connect Inspector](https://developers.onelogin.com/openid-connect/inspector) tool to help you debug. You can use the tool to send requests to OneLogin and check what is sent in the payload. Use the tool to verify you are using the correct claims and scopes. When using this, you will need to add the Inspector callback URL to your application Redirect URIs list. 
[OpenID Connect Inspector Tutorial](https://youtu.be/do0agd71hE8)
```
Language cleanup.

For adding a Youtube video, use the video component. Refer to this part of the [README](https://github.com/spectrocloud/librarium#youtube-video) for example and usage.

# References

> [OneLogin](https://www.onelogin.com/)
> [Documentation](https://developers.onelogin.com/)
> [OpenID Connect Inspector](https://developers.onelogin.com/openid-connect/inspector)

Reviewer: ```suggestion
- [OneLogin](https://www.onelogin.com/)

- [Documentation](https://developers.onelogin.com/)

- [OpenID Connect Inspector](https://developers.onelogin.com/openid-connect/inspector)
```
